### PR TITLE
#373 Resolves immutable updatedAt column

### LINKED ISSUE #373: The updatedAt property of SlurFilter is immutable
Seeing this error in the logs:

```
HHH000502: The [updatedAt] property of the [com.sublinks.sublinksapi.slurfilter.entities.SlurFilter] entity was modified, but it won't be updated because the property is immutable.
```

### PR DESCRIPTION
**Pull request by Pdzly**
closes #373 
( wrongly named branch, please ignore the number)

### CODE REVIEW

File: src/main/resources/db/migration/V20231003__Create_initial_entity_tables.sql [hunk 1]
```diff
@@ -156,7 +156,7 @@ CREATE TABLE link_person_instances
   id          BIGSERIAL PRIMARY KEY,
   person_id   BIGINT                                    NOT NULL,
   instance_id BIGINT                                    NOT NULL,
-  created_at  TIMESTAMP(3) DEFAULT CURRENT_TIMESTAMP(3) NOT NULL
+  created_at  TIMESTAMP(3) DEFAULT CURRENT_TIMESTAMP(3) NULL
```
Comment (jgrim) [inline src/main/resources/db/migration/V20231003__Create_initial_entity_tables.sql:159]: Created at should not be nullable

File: src/main/resources/db/migration/V20231003__Create_initial_entity_tables.sql [hunk 2]
```diff
@@ -178,7 +178,7 @@ CREATE TABLE instance_blocks
 (
   id          BIGSERIAL PRIMARY KEY,
   instance_id BIGINT                                    NOT NULL,
-  created_at  TIMESTAMP(3) DEFAULT CURRENT_TIMESTAMP(3) NOT NULL
+  created_at  TIMESTAMP(3) DEFAULT CURRENT_TIMESTAMP(3) NULL
```
Comment (jgrim) [inline src/main/resources/db/migration/V20231003__Create_initial_entity_tables.sql:181]: Created at should not be nullable

**Review by jgrim — CHANGES_REQUESTED** [review]
Created at should not be nullable

File: src/main/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterService.java [hunk 3]
```diff
@@ -22,7 +22,7 @@ public class SlurFilterService {
   @Transactional
   public void updateOrCreateLemmySlur(String regex) {
 
-    SlurFilter slurFilter = slurFilterRepository.findAll().stream().findFirst().orElse(null);
+    SlurFilter slurFilter = slurFilterRepository.findById(1L).orElse(null);
```
Comment (lazyguru) [inline src/main/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterService.java:25]: What if ID 1 gets deleted? Why not keep it the way it was before and just grab the first record?

File: src/main/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterService.java [hunk 4]
```diff
@@ -36,11 +36,11 @@ public void updateOrCreateLemmySlur(String regex) {
 
   public SlurFilter getLemmySlurFilter() {
 
-    SlurFilter slurFilter = slurFilterRepository.findAll().stream().findFirst().orElse(null);
+    SlurFilter slurFilter = slurFilterRepository.findById(1L).orElse(null);
```
Comment (lazyguru) [inline src/main/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterService.java:39]: Same here

**Review by lazyguru — COMMENTED** [review]
Also... none of the timestamp fields should be nullable

File: src/main/resources/db/migration/V20231003__Create_initial_entity_tables.sql (hunk 1 above)
Comment (Pdzly) [reply src/main/resources/db/migration/V20231003__Create_initial_entity_tables.sql:159]: sadly the DEFAULT goes AFTER the check of NULL in the ORM. But yeah then we just need to make the required to false in the orm.

File: src/main/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterService.java (hunk 4 above)
Comment (Pdzly) [reply src/main/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterService.java:39]: As if its getting deleted lemmy support is dropped. This should ensure, it wont make any side effects.

File: src/main/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterService.java (hunk 3 above)
Comment (Pdzly) [reply src/main/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterService.java:25]: As otherwise side effects could ruin the platform.

File: src/main/resources/db/migration/V20231003__Create_initial_entity_tables.sql (hunk 1 above)
Comment (Pdzly) [reply src/main/resources/db/migration/V20231003__Create_initial_entity_tables.sql]: nvm i had a brainfart somehow there. But it is right that the @UpdatedTimestamp does some weird sh*t and it needs to be required=False in the ORM. But in the sql it can stay.

File: src/main/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterService.java (hunk 4 above)
Comment (jgrim) [reply src/main/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterService.java:39]: I agree, we shouldn't be hard coding db keys.

Comment (Pdzly) [reply src/main/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterService.java:39]: Ok this could harm a lot of a instance if there is an error.

**Comment by urda** [comment]
I'm not sure how I ended up closing this. I must have done it from CLI by mistake.

File: src/main/java/com/sublinks/sublinksapi/slurfilter/entities/SlurFilter.java [hunk 5]
```diff
@@ -47,8 +47,8 @@ public class SlurFilter {
   @Column(updatable = false, nullable = false, name = "created_at")
   private Date createdAt;
 
-  @UpdateTimestamp
-  @Column(updatable = false, nullable = false, name = "updated_at")
```
Comment (jgrim) [inline src/main/java/com/sublinks/sublinksapi/slurfilter/entities/SlurFilter.java:51]: nullable should be false

File: src/main/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterService.java [hunk 6]
```diff
@@ -9,6 +9,8 @@
 import java.util.List;
```
Comment (jgrim) [inline src/main/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterService.java:9]: The changes in this file are unrelated to the original issue. I think they should be moved to a new issue. I have an idea on how to better implement the slur filter lookup.

File: src/test/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterServiceUnitTests.java [hunk 7]
```diff
@@ -22,6 +22,8 @@
 import org.mockito.InjectMocks;
```
Comment (jgrim) [inline src/test/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterServiceUnitTests.java:22]: Same as other comment, this should be moved to a separate issue.

**Comment by jgrim** [comment]
@Pdzly I've thought more about this issue. I think we should remove the db triggers and let springboot manage updatedAt for the entities. Rather than removing the annotation I believe we should make the field mutable and allow springboot to handle it. I'd consider updating the annotations for update and create to be driven by the database timestamp like so:

`@UpdateTimestamp(source = SourceType.DB)` & `@CreationTimestamp(source = SourceType.DB)`

**Comment by jgrim** [comment]
Renamed branch to `issue/373-slur-filter` See new PR: https://github.com/sublinks/sublinks-api/pull/389
